### PR TITLE
Migrate to Xcode 16 minimum

### PR DESCRIPTION
**Pull request by dfeinzimer**
Completed:
- [x] Verified all example projects build okay
- [x] Verified tests
- [x] Verified affected component behavior
- [x] JobManagerExample crashes on launch in iOS 18
- [x] Resolve `TableTopSceneView` crash (Ref Swift 6148)

### CODE REVIEW

File: Sources/ArcGISToolkit/Components/Search/SearchSource.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import ArcGIS
 
 /// Defines the contract for a search result provider.
-public protocol SearchSource {
+public protocol SearchSource: Sendable {
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Search/SearchSource.swift:18]: While adopting the `Sendable` protocol is easy, doing so right can be hard. Neither of the conforming types are actually sendable nor would it be easy to make them so. That is why I didn't do this myself when enabling complete strict concurrency. We may end up having no other choice, but if so, we should make an effort to make the conforming types thread safe, as difficult as that will be. The best option is probably to re-design the API, but I don't think we want to do that right now.

File: Package.swift [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.9
+// swift-tools-version:6.0
```
Comment (philium) [inline Package.swift:1]: Originally the version number needed to be immediately after the colon, but over time the whitespace requirements have been relaxed, allowing for them to be separated with a space.

```suggestion
// swift-tools-version: 6.0
```

File: Examples/ExamplesApp/AnyExample.swift [hunk 3]
```diff
@@ -16,9 +16,9 @@ import SwiftUI
 
 struct AnyExample<Content: View> {
     var name: String
-    var content: @MainActor () -> Content
+    var content: () -> Content
     
-    init(_ name: String, content: @autoclosure @escaping @MainActor () -> Content) {
+    init(_ name: String, content: @autoclosure @escaping () -> Content) {
```
Comment (philium) [inline Examples/ExamplesApp/AnyExample.swift:21]: What's the reason for these changes?

File: Sources/ArcGISToolkit/Components/Authentication/ValueContinuation.swift [hunk 4]
```diff
@@ -28,7 +28,9 @@ import Foundation
     func setValue(_ value: Value) {
         guard _value == nil else { return }
         _value = value
-        continuation?.resume(returning: value)
+        Task {
+            continuation?.resume(returning: value)
+        }
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Authentication/ValueContinuation.swift:33]: Instead of doing this, I suggest adding a requirement that `Value` be sendable.

File: Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentImportMenu.swift [hunk 5]
```diff
@@ -204,7 +204,9 @@ struct AttachmentImportMenu: View {
 #if !targetEnvironment(macCatalyst) && !targetEnvironment(simulator)
             .onCameraCaptureModeChanged { captureMode in
                 if captureMode == .video && AVCaptureDevice.authorizationStatus(for: .audio) == .denied {
-                    microphoneAccessAlertIsVisible = true
+                    Task { @MainActor in
+                        microphoneAccessAlertIsVisible = true
+                    }
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/FeatureFormView/AttachmentHelpers/AttachmentImportMenu.swift:209]: Instead of these changes, annotate the action closure with `MainActor` in the `AttachmentCameraController.onCameraCaptureModeChanged(perform:)` declaration.

File: Examples/ExamplesApp/AnyExample.swift (hunk 3 above)
Comment (dfeinzimer) [reply Examples/ExamplesApp/AnyExample.swift:21]: Leaving these annotations in place produces the following errors in `Examples.swift`.

Screenshot 2024-10-02 at 11 01 38 AM

Given that `View`s are now annotated `@MainActor` in Xcode 16, I don't understand why some of these initializers are erroring while others are not. Despite this, these changes match `AnyExample` in the Swift Examples app.

Comment (philium) [reply Examples/ExamplesApp/AnyExample.swift:21]: I see. I didn't see the errors, but that's because I was building the toolkit itself instead of the Examples app.

File: Sources/ArcGISToolkit/Components/Search/SearchSource.swift (hunk 1 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Search/SearchSource.swift]: I've marked `SearchSource` `@MainActor` and I think it's a suitable solution for now.

Comment (philium) [reply Sources/ArcGISToolkit/Components/Search/SearchSource.swift]: Since you're adding a new concurrency constraint, make sure to add the `preconcurrency` attribute to the declaration.

File: Sources/ArcGISToolkit/Components/OverviewMap.swift [hunk 6]
```diff
@@ -47,7 +47,6 @@ import SwiftUI
 /// the [Examples](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/tree/main/Examples/Examples)
 /// and refer to [OverviewMapExampleView.swift](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/main/Examples/Examples/OverviewMapExampleView.swift)
 /// in the project. To learn more about using the `OverviewMap` see the <doc:OverviewMapTutorial>.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/OverviewMap.swift:50]: This attribute should be removed as well.

File: Sources/ArcGISToolkit/Components/Popups/PopupView.swift [hunk 7]
```diff
@@ -51,7 +51,6 @@ import ArcGIS
 /// and refer to
 /// [PopupExampleView.swift](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/main/Examples/Examples/PopupExampleView.swift)
 /// in the project. To learn more about using the `PopupView` see the <doc:PopupViewTutorial>.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Popups/PopupView.swift:54]: This attribute should be removed as well.

File: Sources/ArcGISToolkit/Components/Scalebar/Scalebar.swift [hunk 8]
```diff
@@ -49,7 +49,6 @@ import SwiftUI
 /// To see it in action, try out the [Examples](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/tree/main/Examples/Examples)
 /// and refer to [ScalebarExampleView.swift](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/main/Examples/Examples/ScalebarExampleView.swift) 
 /// in the project. To learn more about using the `Scalebar` see the <doc:ScalebarTutorial>.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Scalebar/Scalebar.swift:52]: This attribute should be removed as well.

File: Sources/ArcGISToolkit/Components/Search/SearchSource.swift [hunk 9]
```diff
@@ -15,7 +15,7 @@
 import ArcGIS
 
 /// Defines the contract for a search result provider.
-public protocol SearchSource {
+@MainActor public protocol SearchSource {
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Search/SearchSource.swift:18]:
```suggestion
@MainActor
@preconcurrency
public protocol SearchSource {
```

File: Sources/ArcGISToolkit/Components/Search/SearchView.swift [hunk 10]
```diff
@@ -60,7 +60,6 @@ import ArcGIS
 /// To see the `SearchView` in action, and for examples of `Search` customization, check out the [Examples](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/tree/main/Examples/Examples)
 /// and refer to [SearchExampleView.swift](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/main/Examples/Examples/SearchExampleView.swift)
 /// in the project. To learn more about using the `SearchView` see the <doc:SearchViewTutorial>.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Search/SearchView.swift:63]: This attribute should be removed as well.

File: Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift [hunk 11]
```diff
@@ -61,7 +61,6 @@ import SwiftUI
 /// To see the `UtilityNetworkTrace` in action, check out the [Examples](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/tree/main/Examples/Examples)
 /// and refer to [UtilityNetworkTraceExampleView.swift](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/main/Examples/Examples/UtilityNetworkTraceExampleView.swift)
 /// in the project. To learn more about using the `UtilityNetworkTrace` see the <doc:UtilityNetworkTraceTutorial>.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTrace.swift:64]: This attribute should be removed as well.

File: Sources/ArcGISToolkit/Utility/AsyncImageView.swift [hunk 12]
```diff
@@ -16,7 +16,6 @@ import SwiftUI
 import ArcGIS
 
 /// A view displaying an async image, with error display and progress view.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Utility/AsyncImageView.swift:19]: This attribute should be removed as well.

File: Sources/ArcGISToolkit/Utility/SearchField.swift [hunk 13]
```diff
@@ -16,7 +16,6 @@ import SwiftUI
 
 /// A custom view implementing a SearchField. It contains a search button, text field, delete text button,
 /// and a button to allow users to hide/show the search results list.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Utility/SearchField.swift:19]: This attribute should be removed as well.

File: Sources/ArcGISToolkit/Components/Search/SearchSource.swift (hunk 9 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Search/SearchSource.swift:18]: Why is `@preconcurrency` needed here?

Comment (philium) [reply Sources/ArcGISToolkit/Components/Search/SearchSource.swift]: Because a new concurrency constraint is being added. Without it, if a consumer hasn't adopted complete strict concurrency checking, they will get a warnings (Swift 5 language mode) or an error (Swift 6 language mode).

Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Search/SearchSource.swift]: Makes sense, thank you.

**Comment by dfeinzimer** [comment]
Marking this as ready for review. There is one crash with `TableTopSceneView` that is yet to be resolved.

**Review by mhdostal — DISMISSED** [review]

**Comment by philium** [comment]
> There is one crash with `TableTopSceneView` that is yet to be resolved.

Should we comment out the implementation of those two delegate methods for now? Then I think this PR could be merged.

**Comment by dfeinzimer** [comment]
> Should we comment out the implementation of those two delegate methods for now?

@rolson / @mhdostal thoughts? I'm okay to wait a bit to get a response on today's new support ticket.

**Comment by rolson** [comment]
> > Should we comment out the implementation of those two delegate methods for now?
> 
> @rolson / @mhdostal thoughts? I'm okay to wait a bit to get a response on today's new support ticket.

Assuming that commenting out those delegate methods affect or limit previously released functionality, then no we should not comment them out. I don't want to merge this until we either:

1. have a workaround
2. go back to swift 5 mode
3. make the call to not support whatever functionality is affected by removing these delegate methods

**Comment by dfeinzimer** [comment]
The `TableTopSceneView` workaround is implemented. This is ready for review once again.

File: Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift [hunk 14]
```diff
@@ -114,11 +116,24 @@ extension ARSwiftUIView: UIViewRepresentable {
 
 extension ARSwiftUIView {
     class Coordinator: NSObject, ARSCNViewDelegate, ARSessionDelegate {
+        struct State: Sendable {
```
Comment (rolson) [inline Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift:119]:
```suggestion
        private struct State: Sendable {
```

File: Sources/ArcGISToolkit/Components/Augmented Reality/TableTopSceneView.swift [hunk 15]
```diff
@@ -17,7 +17,6 @@ import SwiftUI
 import ArcGIS
 
 /// A scene view that provides an augmented reality table top experience.
-@MainActor
 @preconcurrency
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Augmented Reality/TableTopSceneView.swift:20]: Please remove this attribute as well.

File: Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift [hunk 16]
```diff
@@ -114,11 +116,24 @@ extension ARSwiftUIView: UIViewRepresentable {
 
 extension ARSwiftUIView {
     class Coordinator: NSObject, ARSCNViewDelegate, ARSessionDelegate {
+        private struct State: Sendable {
+            var onAddNodeAction: (@MainActor (SceneParameters) -> Void)?
+            var onUpdateNodeAction: (@MainActor (SceneParameters) -> Void)?
+        }
+        
+        private let state = OSAllocatedUnfairLock(initialState: State())
```
Comment (philium) [inline Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift:124]: This is overkill. Instead, just refactor `Coordinator` to make all of its properties immutable. While you're at it, I suggest moving `Coordinator` into the previous extension (the one for `UIViewRepresentable`, since `Coordinator` satisfies a requirement of that protocol):

```swift
func updateUIView(_ uiView: ARViewType, context: Context) {}

func makeCoordinator() -> Coordinator {
    return .init(
        onDidChangeGeoTrackingStatusAction: onDidChangeGeoTrackingStatusAction,
        onCameraDidChangeTrackingStateAction: onCameraDidChangeTrackingStateAction,
        onDidUpdateFrameAction: onDidUpdateFrameAction,
        onAddNodeAction: onAddNodeAction,
        onUpdateNodeAction: onUpdateNodeAction
    )
}

class Coordinator: NSObject {
    let onDidChangeGeoTrackingStatusAction: ((ARSession, ARGeoTrackingStatus) -> Void)?
    let onCameraDidChangeTrackingStateAction: ((ARSession, ARCamera.TrackingState) -> Void)?
    let onDidUpdateFrameAction: ((ARSession, ARFrame) -> Void)?
    let onAddNodeAction: (@MainActor (SceneParameters) -> Void)?
    let onUpdateNodeAction: (@MainActor (SceneParameters) -> Void)?
    
    init(
        onDidChangeGeoTrackingStatusAction: ((ARSession, ARGeoTrackingStatus) -> Void)?,
        onCameraDidChangeTrackingStateAction: ((ARSession, ARCamera.TrackingState) -> Void)?,
        onDidUpdateFrameAction: ((ARSession, ARFrame) -> Void)?,
        onAddNodeAction: (@MainActor (SceneParameters) -> Void)?,
        onUpdateNodeAction: (@MainActor (SceneParameters) -> Void)?
    ) {
        self.onDidChangeGeoTrackingStatusAction = onDidChangeGeoTrackingStatusAction
        self.onCameraDidChangeTrackingStateAction = onCameraDidChangeTrackingStateAction
        self.onDidUpdateFrameAction = onDidUpdateFrameAction
        self.onAddNodeAction = onAddNodeAction
        self.onUpdateNodeAction = onUpdateNodeAction
    }
}
```

and then conforming it to `ARSCNViewDelegate` and `ARSessionDelegate` in extensions:

```swift
extension ARSwiftUIView.Coordinator: ARSCNViewDelegate {
    func renderer(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
        let sceneParameters = SceneParameters(renderer: renderer, node: node, anchor: anchor)
        Task { @MainActor [onAddNodeAction] in
            onAddNodeAction?(sceneParameters)
        }
    }
    
    func renderer(_ renderer: SCNSceneRenderer, didUpdate node: SCNNode, for anchor: ARAnchor) {
        let sceneParameters = SceneParameters(renderer: renderer, node: node, anchor: anchor)
        Task { @MainActor [onUpdateNodeAction] in
            onUpdateNodeAction?(sceneParameters)
        }
    }
}

extension ARSwiftUIView.Coordinator: ARSessionDelegate {
    func session(_ session: ARSession, didChange geoTrackingStatus: ARGeoTrackingStatus) {
        onDidChangeGeoTrackingStatusAction?(session, geoTrackingStatus)
    }
    
    func session(_ session: ARSession, cameraDidChangeTrackingState camera: ARCamera) {
        onCameraDidChangeTrackingStateAction?(session, camera.trackingState)
    }
    
    func session(_ session: ARSession, didUpdate frame: ARFrame) {
        onDidUpdateFrameAction?(session, frame)
    }
}
```

Note that `MainActor.run` isn't necessary either.

File: Sources/ArcGISToolkit/Components/Augmented Reality/TableTopSceneView.swift (hunk 15 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Augmented Reality/TableTopSceneView.swift:20]: This is confusing, this attribute [is not there](https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/blob/27ce6fc7e2c5fcd61a62236e1590df1f7bba074e/Sources/ArcGISToolkit/Components/Augmented%20Reality/TableTopSceneView.swift#L20-L21) and the feedback comment doesn't have a yellow "Outdated" tag. 😆

File: Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift (hunk 16 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/ARSwiftUIView.swift]: Applied, thank you!

**Review by philium — APPROVED** [review]

**Review by mhdostal — APPROVED** [review]
